pkg/apis/query/v1alpha1/cache_types.go

```diff
@@ -0,0 +1,151 @@
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// ResourceKindClusterCache is kind name of ClusterCache.
+	ResourceKindClusterCache = "ClusterCache"
+	// ResourceSingularClusterCache is singular name of ClusterCache.
+	ResourceSingularClusterCache = "clusterCache"
+	// ResourcePluralClusterCache is plural name of ClusterCache.
+	ResourcePluralClusterCache = "clusterCaches"
+	// ResourceNamespaceScopedClusterCache indicates if ClusterCache is not NamespaceScoped.
+	ResourceNamespaceScopedClusterCache = false
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:shortName=cc
+
+// ClusterCache defines a list of member cluster to be cached.
+type ClusterCache struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec represents the desired behavior of ClusterCache.
+	Spec ClusterCacheSpec `json:"spec,omitempty"`
+
+	// Status represents the status of ClusterCache.
+	// +optional
+	Status ClusterCacheStatus `json:"status,omitempty"`
+}
+
+// ClusterCacheSpec defines the desired state of ClusterCache
+type ClusterCacheSpec struct {
+	// ClusterSelectors represents the filter to select clusters.
+	// +required
+	ClusterSelectors []ClusterSelector `json:"clusterSelectors"`
+
+	// ResourceSelectors used to select resources.
+	// +required
+	ResourceSelectors []ResourceSelector `json:"resourceSelectors"`
```

Comment:
I'm not familiar with the index thing. I wonder to know do we usually set up `index`s on specific `kinds` or not?
In other words, is there any difference between the index for `Pods` and `Secret`? If yes, the `ResourceSelector` more like a `rules` about how to select resources and how to set up the index.

Reply:
I'm not sure if the user is required to decide the resource scope.